v2/tools/generator/internal/codegen/pipeline/add_secrets.go

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package pipeline
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+
+	"github.com/Azure/azure-service-operator/v2/tools/generator/internal/astmodel"
+	"github.com/Azure/azure-service-operator/v2/tools/generator/internal/config"
+)
+
+// AddSecretsStageID is the unique identifier for this pipeline stage
+const AddSecretsStageID = "addSecrets"
+
+// AddSecrets replaces properties flagged as secret with genruntime.SecretReference
+func AddSecrets(_ *config.Configuration) Stage { // TODO: Will need this configuration parameter eventually to allow config overrides
```

Comment:
Doesn't this stage have pre/posrequisites that should be explicit?

At the very least, the stages creating the ARM types must run second, yes?

Reply:
Yeah it has to run between the SpecWithStatus stage and the ARM types stage. I've added:
```
RequiresPrerequisiteStages(AugmentSpecWithStatusStageID).RequiresPostrequisiteStages(CreateARMTypesStageID)
```